### ISSUE TITLE
Course progress block should be visible to enrolled users only, by default

### ISSUE THREAD

**Issue by eri-trabiccolo** [gocodebox/lifterlms-blocks#44]
### 1. Reproduction Steps

+ Create course using the block editor
+ Visit the course page as visitor

### 2. Expected Behavior

+ I would expect to NOT see the course progress bar

### 3. Actual Behavior

+ I can see it

### 4. Error Messages / Logs

n/a

### 5. System Report

LifterLMS: 3.37.16

LifterLMS Blocks: 1.6.0

### 6. Browser, Device, and Operating System

n/a

**Comment by thomasplevy** [comment]
https://github.com/gocodebox/lifterlms/blob/11dd83683671539bfeafb0d58194074c3bcfa81e/includes/shortcodes/class.llms.shortcodes.php#L399

Shortcode should have an attribute to "check_enrollment"

When output by the block we should have "check_enrollment" true -- 

If current user is enrolled, output the block, otherwise do not

**Comment by thomasplevy** [comment]
Note: this change will affect the progress bar widget too -- add a widget option to allow changing default behavior as neede

**Comment by thomasplevy** [comment]
@eri-trabiccolo noticed that this is something of a duplicate of #23 

I was about to start working on that and noticed this one and that it's assigned to you. Have you begun work on this? If not I'm going to tackle it while I'm working on block things.

**Comment by eri-trabiccolo** [comment]
@thomasplevy not started yet no :/

**Comment by thomasplevy** [comment]
I'm gonna work on this right now then

**Comment by eri-trabiccolo** [comment]
@thomasplevy and I think we should blacklist this block for the enrollment visibility settings, makes no sense - it can only be visible for enrolled users (which includes logged in) right?!

**Comment by thomasplevy** [comment]
After a full day refactor I realized that this has nothing to do with #23

**Comment by thomasplevy** [comment]
@eri-trabiccolo I didn't get to this so you can hop back on if you want, otherwise I'll pick it back up on Monday, I'm going to stick with these block editor bugs for a few more days I guess...

**Comment by thomasplevy** [comment]
> @thomasplevy and I think we should blacklist this block for the enrollment visibility settings, makes no sense - it can only be visible for enrolled users (which includes logged in) right?!

We discussed how, currently, the shortcode doesn't actually "hide" the progress for non-enrolled users and if we automatically did hide it it could cause issues...

See the comment above about modifying the shortcode and actually allowing visibility anyway

**Comment by eri-trabiccolo** [comment]
@thomasplevy 
mmm the thing is that according to this:

> Shortcode should have an attribute to "check_enrollment"
> 
> When output by the block we should have "check_enrollment" true 

This means that this way the block will already ALWAYS instruct the shortcode to print the progress bar only for enrolled users, so the visibility settings make just no sense, to my understanding. So with the above we do automatically hide it.

We probably didn't stay we should not allow enrollments visibility settings for this, but it's only a logical consequence to me...

**Comment by thomasplevy** [comment]
Fixed by #58

**Comment by eri-trabiccolo** [comment]
well fixed by #58 and https://github.com/gocodebox/lifterlms/pull/1125 once merged...

**Comment by eri-trabiccolo** [comment]
@thomasplevy 
I need to reopen this because I think I made an error:
In my latest "rework" I switched the block back to a non-dynamic block, this means that the change we made won't be taken into account unless you save the course.
So what do we do now?
1) We accept this because "hey this is normal we don't want to change the behavior you are used to"
2) We make it dynamic again and add a new deprecated version (so we test it too :D) because we've already released this publicly.
